src/Bicep.Core/TypeSystem/DeployTimeConstantVisitor.cs

```diff
@@ -191,20 +213,48 @@ public override void VisitPropertyAccessSyntax(PropertyAccessSyntax syntax)
                 // nested errorSyntax up the stack.
                 this.errorSyntax = syntax;
             }
-            else if (syntax.BaseExpression is VariableAccessSyntax variableAccessSyntax)
+            else
             {
-                // This is a non-overlapping error, which means that there's two or more runtime properties being referenced
-                if (this.errorSyntax != null)
-                {
-                    this.AppendError();
-                }
-                if (ExtractResourceOrModuleSymbolAndBodyObj(this.model, variableAccessSyntax) is ({ } declaredSymbol, { } referencedBodyObj) &&
-                    referencedBodyObj.Properties.TryGetValue(syntax.PropertyName.IdentifierName, out var propertyType) &&
-                    !propertyType.Flags.HasFlag(TypePropertyFlags.DeployTimeConstant))
+                switch (syntax.BaseExpression)
                 {
-                    this.errorSyntax = syntax;
-                    this.accessedSymbol = declaredSymbol.Name;
-                    this.referencedBodyObj = referencedBodyObj;
+                    case VariableAccessSyntax variableAccessSyntax:
```

Comment:
Is it possible to consolidate these branches?

Reply:
Good pt. Should be possible to extract a common method.

Reply:
Done